src/NServiceBus.PersistenceTests/Sagas/When_worker_tries_to_complete_saga_update_by_another_optimistic.cs

```diff
@@ -42,7 +42,7 @@ public async Task Should_fail()
         }
         finally
         {
-            winningSaveSession.Dispose();
+            await winningSaveSession.DisposeAsync();
```

Comment:
This change aligns with the test's spirit as is, but it's unclear why a using block wasn't used instead of try/finally. (same comment on the next change as well)

Reply:
Excellent question! Originally, the intent was to use the using blocks wherever it makes sense and when explicit control is needed, use the try finally approach. This test is a weird in-between thing that could benefit for the winning session to be wrapped in a using, since it would probably even improve the readability of this test.

Reply:
Will be addressing this in a separate PR